addons/web/static/src/core/ui/ui_service.js

```diff
@@ -201,12 +229,16 @@ export const uiService = {
             get isBlocked() {
                 return blockCount > 0;
             },
+            get externalWindow() {
+                return externalWindow && externalWindow.closed ? null : externalWindow;
+            },
             isSmall: utils.isSmall(),
             block,
             unblock,
             activateElement,
             deactivateElement,
             getActiveElementOf,
+            getExternalWindow,
```

Comment:
I would not add this to the UI service (not now at least), as there's a single usecase (and because it's debatable that such a feature should be in there). So for now, I would keep this in mail, alongside the feature using it

Reply:
But I'm using the `bus` of the service, would it not make sense to keep it here?

Reply:
Why do you trigger a resize event?

Reply:
To re-render the form view since the position of the chatter/document viewer is determined at view compile time

Reply:
Ok I didn't read that part of the code... You add the listener, so you can basically do whatever you want, i.e. trigger any event on any bus you want. You add your own service in mail for that popup window, and have a bus there, and trigger an event (why "resize"??) on that bus.

Reply:
I extracted that in a new service declared in `mail`

The resize is because the space available on the screen changes.
All the listeners to know how to display are already tied to that event to know the layout of the page.

Reply:
> You add the listener

No, there are already other listeners too.
Like https://github.com/odoo-dev/enterprise/blob/d433b1518aab1b04c320b3349dc74fc6cd9e618d/account_accountant/static/src/components/move_line_list/move_line_list.js#L32